### DISCUSSION TITLE
Backup and Restore Diskernet

### DISCUSSION THREAD

**Discussion by abenezertasissa** [c9fe/22120#109] [Q&A] [answered]
Is there a feature where we can backup offline pages to use on a different computer? let's say you formatted your PC and want to get your archive back?

What happens if a site gets updated? would the previous content be overwritten?

**Comment by crisdosaygo** [comment, marked as answer]
Hey, thanks for asking that.

Yes, so--the archive is simply a directory subtree. That means it is just a root directory (listed on the settings page, that you can change), and a set of subdirectories and files. That is your entire archive (for that root directory ~~ when you change the root directory you create an new archive).

So, that makes it *very simply* to transfer your archive to a new machine. Just "zip up" (using WinZip, or whatever compression utility you like) your archive folder (or, equivalently, "sync" it to Dropbox, Box, Drive, the Windows one (OneDrive, heh ~~ forgot for a second) and you can copy and transfer it wherever you like. 

Another thing is...the archive is "git diffable". So you can run `git init` inside an archive root and periodically commit the changes to the git history. This enables you to "roll back" and "roll forward" to any point in the history of your archive. If `git` is not your thing, there are other methods of version control you could use. 

Also, to answer your 2nd question, yes, for a given archive, when you revisit a site (a page URL) we "merge" the content of any resources used by that page. That sounds more complex than it is -- all I mean is that, any new resources the page uses that time are added, but all previous ones are kept. The effect of this is a little subtle, so there should be a somewhat more lengthy explanation.

I'm sorry, please bear with it: say on Jan 1 you requested https://example.com/image-carousel/index.html and you got delivered (as part of loading and saving that page) images for products A, B and C. But when you went back on Feb 1, you got delivered images for products B, C, D, E, and F.

What happens in your archive? Your archive contains the saved copies of products A through F.

But now, there's two things that can happen, that depend entirely on the page you saved. Let's assume you are travelling through the multiverse and are able to simultaneously view these two different possible futures:

1. In Universe 1: You will be unable to access A because the page no longer contains any references to it. For example: say the image carousel on Feb 1 omitted any links or images for product A's image. Product A is still on your disk, but you cannot access it from the Feb 1 archived copy of  index.html because that copy has no links to A. 
2. In Universe 2: You will still be able to access A because the page still contains a link to A (such as an image or a link to its 'product page' or whatever). This is because the Feb 1 copy of index.html (in Universe 2) still has a link to A. 

So, DK (diskernet) / 22120 currently has not way to do any snapshotting or "time capsuling" of the archive data. But you can still access both copies of A if you use a version control system (like git), so long as you committed any changes to your archive git repository on both Jan 1, and Feb 1, then you can use git to roll between those two, and when you have "checked out" the snapshot you want, just use DK/22120 to view it as normal. 😄 

Hope that helps! :joy: Have a great day and thank you for your question, @abene2
